decimal_test.go

```diff
@@ -191,20 +191,35 @@ func TestNewFromFloatWithExponent(t *testing.T) {
 		float float64
 		exp   int32
 	}
+	// some tests are taken from here https://www.cockroachlabs.com/blog/rounding-implementations-in-go/
 	tests := map[Inp]string{
-		Inp{123.4, -3}:      "123.4",
-		Inp{123.4, -1}:      "123.4",
-		Inp{123.412345, 1}:  "120",
-		Inp{123.412345, 0}:  "123",
-		Inp{123.412345, -5}: "123.41235",
-		Inp{123.412345, -6}: "123.412345",
-		Inp{123.412345, -7}: "123.412345",
+		Inp{123.4, -3}:                 "123.4",
+		Inp{123.4, -1}:                 "123.4",
+		Inp{123.412345, 1}:             "120",
+		Inp{123.412345, 0}:             "123",
+		Inp{123.412345, -5}:            "123.41235",
+		Inp{123.412345, -6}:            "123.412345",
+		Inp{123.412345, -7}:            "123.412345",
+		Inp{123.9999999999999999, -7}:  "124",
+		Inp{123.8989898999999999, -7}:  "123.8989899",
+		Inp{0.49999999999999994, 0}:    "0",
+		Inp{0.5, 0}:                    "1",
+		Inp{0.5000000000000001, 0}:     "1",
+		Inp{1.390671161567e-309, 0}:    "0",
+		Inp{4.503599627370497e+15, 0}:  "4503599627370497",
+		Inp{4.503599627370497e+60, 0}:  "4503599627370497110902645731364739935039854989106233267453952",
```

Comment:
This is because of internal float representation.
Look here https://play.golang.org/p/dMKlut2Hw4R